src/python/bcc/table.py

```diff
@@ -922,3 +947,48 @@ def __init__(self, *args, **kwargs):
 class MapInMapHash(HashTable):
     def __init__(self, *args, **kwargs):
         super(MapInMapHash, self).__init__(*args, **kwargs)
+
+class RingBuf(EventArrayBase):
```

Comment:
RingBuf has nothing in common with EventArrayBase, ringbuf is not an array, it doesn't share any functionality. I think this is just plain wrong and misleading.

Reply:
What would you rename it to? EventMapBase? EventTableBase?

Reply:
Or maybe EventBufferBase?

Reply:
I don't know if we need a common Base? What's to be shared in terms of functionality?

Reply:
@anakryiko The common base is for resolving event types. See the methods exposed by EventArrayBase. Before this was all part of PerfEventArray, but keeping it that way would result in code duplication.

Reply:
You mean _get_event_class() stuff? Couldn't that be just a static helper method (or whatever Python equivalent of that)? EventArrayBase doesn't make any sense whatsoever for ringbuf, because it neither is an array-like map, nor it deals with perf events.

Reply:
@anakryiko Fair enough, that sounds like a good solution.